core/src/validator.rs

```diff
@@ -1658,18 +1660,29 @@ impl Validator {
         *start_progress.write().unwrap() = ValidatorStartProgress::Running;
         if config.use_tpu_client_next {
             if let Some(json_rpc_service) = &json_rpc_service {
-                key_notifies.push(json_rpc_service.get_client_key_updater())
+                key_notifiers.write().unwrap().add(
+                    KeyUpdaterType::RpcService,
+                    json_rpc_service.get_client_key_updater(),
+                );
             }
             // note, that we don't need to add ConnectionClient to key_notifiers
             // because it is added inside Tpu.
         }
 
+        // add connection_cache because it is still used in Forwarder.
```

Comment:
This is unnecessary because when `ConnectionCache` is used in `ForwardingStage`, it is added to the updaters there.

Reply:
looking. maybe due to merge. I saw it was added to the array before here.

Reply:
Yes, it is the same connection cache, we only need to have the key updater once for the same instance. removed this code